app/views/admin/languages/new.slim

```diff
@@ -0,0 +1,7 @@
+h1 Add language to #{@country.name}
+
+= form_for [:admin, @country, @language] do |f|
```

Comment:
@iandouglas @AliSchlereth I'm unsure if this is going to be the proper way to go about this, since not only will I need the ability to assign languages to countries, so I have a namespaced route in admin for that. But I don't need to technically create a new language, so I don't need the language data in the form. It'll just be a dropdown of languages that have been added. I'm currently using the `/admin/countries/:country_id/languages/new` route for this purpose, but is there a better way to set this up? Rather than, seemingy hijacking the new/create functionality when really what I'm doing is amending `country.languages`, especially considering that once I build out functionality to add/delete languages in general, I will need this route? or it would have to be on a different route. Or something.

Reply:
I'm not sure I understand the functionality you are trying to build here. Are you wanting to add a language to a country? I would want a language dropdown on the country new/edit view. The languages/new should be used for creating a new language in general. Let me know if I missed the point of your question.

Reply:
My idea is that on the country edit page I will have a list of the languages that belong to that country. You can click on an "Add Language" button to go to a page with a dropdown, where you can select a language, click "Add" and it sends you back to the edit country page where the new language will now be present.

Alternatively, I would rather do it so you can click "add item" and a dropdown shows up on the same page, and you can add it right from there. Though, I'm not entirely sure how to test javascript functionality with capybara so that's something I'd have to explore.

Reply:
@AliSchlereth While that would work, my data set has 113 unique languages so that might get a bit messy. A dropdown would be preferable. I have the form built out, and it does add languages properly to the country. I guess the question is, should I have a custom post/put handler for this? somthing like, `Admin::LanguagesHandler#add` that could take a `:country_id` and `:language_id` to assign the language to a country? Or is there a proper way to achieve this? A RESTful way to link two datasets together, since Countries->Languages is a many-many relationship

Reply:
@kolyaventuri Hmm yeah, I'd say best user experience might be to have a search bar in your form, but that would require JavaScript and a whole bunch more work. 

To do it without JS, a restful way to work through it might be a new/create on a CountryLanguagesController.  

Have you thought about how you will handle a scenario where a user changes other data on the edit form, clicks out to also change the language, and then wants to keep editing? You'll lose that initial information changed in the original edit form since they don't submit the form and go to an update.